maven-extension/src/main/java/io/opentelemetry/maven/OpenTelemetrySdkService.java

```diff
@@ -106,7 +107,12 @@ public void initialize() {
             .getBoolean("otel.instrumentation.maven.mojo.enabled");
     this.mojosInstrumentationEnabled = mojoSpansEnabled == null ? true : mojoSpansEnabled;
 
-    this.tracer = this.openTelemetry.getTracer("io.opentelemetry.contrib.maven");
+    TracerBuilder tracerBuilder = openTelemetry.tracerBuilder("io.opentelemetry.contrib.maven");
+    String otelMavenExtensionVersion = this.getClass().getPackage().getImplementationVersion();
+    if (otelMavenExtensionVersion != null) {
+      tracerBuilder.setInstrumentationVersion(otelMavenExtensionVersion);
+    }
+    this.tracer = tracerBuilder.build();
```

Comment:
Do we need the `null` check? From looking at the implementation it appears we can go with:

```
this.tracer = this.openTelemetry.getTracer("io.opentelemetry.contrib.maven", this.getClass().getPackage().getImplementationVersion());
```

Reply:
I initially started like this as `implementationVersion`  will only be null in unit tests.

However, I hardened the code because null is a violation of the contract`OpenTelemetry#getTracer(@Nonnull instrumentationName, @Nonnull instrumentationVersion)`.

It's probably an acceptable violation of the contract as the implementation supports null and we would not be afraid of regressions if the implementation starts rejecting null instrumentationVersion because it would fail in unit tests.

Reply:
@kenfinnigan please review the proposed code change. Slightly different from your suggestion in order to also output the otel maven extension version in the logs.